tecken/settings.py

```diff
@@ -584,6 +576,7 @@ def filter(self, record):
 SYMBOL_URLS = _config(
```

Comment:
This is pretty different from how we did Eliot, Antenna, and Socorro where we determine classes and configuration in the settings. For example, with Socorro, we have this:

https://github.com/mozilla-services/socorro/blob/main/socorro/mozilla_settings.py

One of the things I was hoping to do here was re-use as much of that backend infrastructure in Tecken so we had more code/architecture reuse across crash ingestion services.

Can you walk me through your thinking for how backends get specified and configured?

Reply:
The way backends are currently configured isn't great. I had two attempts at reworking that, but ultimately decided to stick with the existing approach to keep the amount of changes smaller.

I looked at the way Socorro does the configuration, but I couldn't generalize that to the case of an arbitrary-length list of storage backends that we have in Tecken. Another difference to our other serivces is that the Tecken buckets are public, so we also need to define how they get accessed from the outside, in addition to defining how they get accessed by Tecken itself. And we also want to support the case that Tecken only accesses the bucket anonymously (bug-1905455).

Ideally, I'd like to be able to provide some structured configuration for storage backends, e.g.
```yaml
upload_backend:
  storage_url: gs://moz-fx-symbols-prod-stage-tecken/
  public_url: https://symbols.cdn.mozilla.net/
  endpoint_url: https://storage.googleapis.com/
try_backend:
  storage_url: gs://moz-fx-symbols-prod-stage-tecken/try
  public_url: https://symbols.cdn.mozilla.net/
  endpoint_url: https://storage.googleapis.com/try
download_backends:
  - storage_url: s3://org.mozilla.crash-stats.symbols-public/
    endpoint_url: https://s3.us-west-2.amazonaws.com/
    anonymous: true
```
This would make both the code and the configuration simple and easy to understand, but it would neither be similar to what Tecken did so far nor to what our other services do.

Do you have a suggestion how to represent a "list of dicts" in a way that's similar to how our other services configure storage?

Reply:
The PR now implements a configuration model that's similar to what Socorro does.